xapian-letor/bin/xapian-prepare-trainingfile.cc

```diff
@@ -75,11 +76,10 @@ try {
 		have_database = true;
 		break;
 	    case 'm':
-		unsigned int size;
-		if (!parse_unsigned(optarg, size)) {
-		    throw "Mset size must be >= 0";
+		if (!parse_unsigned(optarg, msize)) {
+		    cerr << "Mset size must be >= 0" << endl;
 		}
-		msize = size;
+		exit(1);
```

Comment:
This will always exit if `-m` is used, not only when there's an error parsing the argument.  Same problem below too.

Reply:
oh sorry stupid mistake.